pkg/codegen/dotnet/utilities.go

```diff
@@ -118,3 +119,23 @@ func makeSafeEnumName(name, typeName string) (string, error) {
 
 	return safeName, nil
 }
+
+// Provides code for a method which will be placed in the program preamble if deemed
+// necessary. Because many Terraform functions are complex, it is much prettier to
+// encapsulate them as their own function in the preamble.
+func getHelperMethodIfNeeded(functionName string) (string, bool) {
+	var methodBody string
+
+	switch functionName {
+	case "sha1":
+		methodBody =
+			`private static string ComputeSHA1(string input) {
+		return BitConverter.ToString(
+			SHA1.Create().ComputeHash(Encoding.UTF8.GetBytes(%v))
```

Comment:
Is the `%v` intentional here?  It looks like this should be referencing `input`?

Reply:
Done.